### DISCUSSION TITLE
Training my own model

### DISCUSSION THREAD

**Discussion by mazatov** [PaddlePaddle/PaddleOCR#7905] [General]
I'm trying to train my own model that recognizes the 1-2 digit numbers. I have my own dataset with label and bounding boxes around each number. I tried training using the default `det_mv3_db.yml` config file but the text detection results were surprisingly bad. 

So I assume I need to change a few things in the config file, especially in the Train and Test groups. The main difference from the ICDAR is that my images are pretty small. (50-100 pixels). So I'm guessing some parameters need to change. Are there any guidelines on how to set up config file for you own dataset?

**Comment by LDOUBLEV** [comment]
It would be great if you could post a sample from your dataset.

If the number area is obvious in the image, then there is no need to train a text detection model to detect text. It is recommended to train a text recognition model directly.
example：
https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/doc/imgs_words_en/word_401.png

**Reply by mazatov** [reply]
Here's an example of an image in my dataset. So we're taking a crop of the bounding box around the player with a visible jersey number on the back. Realistically it means the numbers are taking ~20% of the image tops. 

That's why I thought training a text detector was also important

![Player bounding box](https://user-images.githubusercontent.com/29291003/199178646-7cf12eea-5e78-4d4d-94be-f4d8f105492b.jpg)

**Reply by gavinfaimdata** [reply]
How big is your dataset? This looks like a task that db should be able to handle if you have enough data. At least for this resolution/quality of image.

**Reply by mazatov** [reply]
I have a pretty big dataset ( > 100K) of similar resolutions. What do you mean by `db` ?

Another potential issue I'm worried about are other symbols that might appear in the crop that are not the jersey number. For example an add under the number or scoreline or something else. Would the recognizer learn to focus only on that particular part of the image?

**Reply by gavinfaimdata** [reply]
You wrote that you're using the config file "det_mv3_db.yml". db is the text detection algorithm you're using.

I just trained a model using db with similarly sized images and had similar results. It seems like it wasn't training on the smaller images, although I could be wrong. I resized them all to 640*640 and just started the training again.

If you have 100k+ images you shouldn't have to worry about your model picking up random ads. Unless the ad looks like the back of a player's jersey it's unlikely your model will mistake it.

How did you get 100k+ labelled images like this? That seems like an incredibly high number.

**Reply by LDOUBLEV** [reply]
Maybe it's an image resolution problem.

The default resolution is 640 when training. However, your training data are pretty small. (50-100 pixels).

Try modifying the part about the input size in the config file. https://github.com/PaddlePaddle/PaddleOCR/blob/5f06a8068eed3e1755b8ca56004cd46c95ac268d/configs/det/det_mv3_db.yml#L76-L84

try:
training part:
      - IaaAugment:
          augmenter_args:
            - { 'type': Fliplr, 'args': { 'p': 0.5 } }
            - { 'type': Affine, 'args': { 'rotate': [-10, 10] } }
      Resize:
          size: [320, 320]

eval part:
      - DetResizeForTest:
          image_shape: [320, 320]

**Reply by gavinfaimdata** [reply]
I just tried resizing all the images in my preprocessing step to 640^2. Hopefully it solves this issue. If not I'll try making the adjustments to the config file

**Reply by gavinfaimdata** [reply]
To get good results I had to resize the image in main() in predict_system.py

**Reply by MrAsifKhan** [reply]
@gavinfaimdata did you get good results when you trained the detection model with 640x640?? 
I am trying to train ResNet50 with SAST with image size of 128*128 and i am getting really bad results.

**Reply by gavinfaimdata** [reply]
I ended up using 320*320 in the end and the results are decent... not as good as I want.

**Reply by camcomharsh** [reply]
increase your resize to [960, 960] and add more augmentation as you are already using mobile net architecture and if possible you should try exploring resnet based architectures
